core/iwasm/aot/aot_runtime.c

```diff
@@ -1621,13 +1744,29 @@ init_func_ptrs(AOTModuleInstance *module_inst, AOTModule *module,
     /* Set import function pointers */
     func_ptrs = (void **)module_inst->func_ptrs;
     for (i = 0; i < module->import_func_count; i++, func_ptrs++) {
-        *func_ptrs = (void *)module->import_funcs[i].func_ptr_linked;
-        if (!*func_ptrs) {
-            const char *module_name = module->import_funcs[i].module_name;
-            const char *field_name = module->import_funcs[i].func_name;
-            LOG_WARNING("warning: failed to link import function (%s, %s)",
-                        module_name, field_name);
+        AOTFunctionInstance *func =
+            aot_locate_import_function_instance(module_inst, i);
+        bh_assert(func->is_import_func);
+
+        if (func->call_conv_wasm_c_api) {
+            /* when execution, goes to invoke_native() and use
+             * c_api_func_imports*/
+            continue;
+        }
+
+        if (func->import_module_inst) {
+            /* when execution, goes to invoke_native() and switch to another
+             * .wasm */
+            continue;
         }
+
+        AOTImportFunc *import_func = func->u.func_import;
+
+        LOG_DEBUG("use wasm_native linked functions for (%s,%s)",
+                  import_func->module_name, import_func->func_name);
+
+        *func_ptrs = import_func->func_ptr_linked;
+        bh_assert(*func_ptrs);
```

Comment:
Per my understanding, the import function doesn't have to be linked and runtime can print warning here, and during execution, the aot code or runtime throws exception "failed to call unlinked import function". The original behavior is like that, we had better not change it?

Reply:
That brings convenience to developers and is a common handle in many VMs like jvm. In many cases the unlinked import function isn't called and developer can still call other functions. No developer reports it a problem, I strongly suggest to keep this. If you want to fix this, had better open another PR to let others to review/discuss and not fix it in this PR.

Reply:
> In many cases the unlinked import function isn't called and developer can still call other functions.

However, this does not align with the specification definition. I believed the purpose of creating the "instantiation linking" series of pull requests was to align with the specification requirements. Otherwise, our main branch code is functioning quite well.